worker/provisioner/provisioner_task.go

```diff
@@ -988,16 +1011,15 @@ func (task *provisionerTask) startMachine(
 	if err != nil {
 		return err
 	}
-
 	startInstanceParams, err := task.setupToStartMachine(machine, v)
 	if err != nil {
-		// Attempt to future proof the AZMap with call to
-		// removeMachineFromAZMap(), and prevent the map becoming
-		// inconsistent
-		task.removeMachineFromAZMap(machine)
 		return task.setErrorStatus("%v", machine, err)
 	}
 
+	if err := task.populateExcludedMachines(machine.Id(), startInstanceParams); err != nil {
```

Comment:
perhaps a comment?
// figure out if the zones available to use for a new instance are restricted based on constraints, and if so exclude those machines from being started in any other zone

Reply:
added.